runtime/hashing/jvm/src/aws/smithy/kotlin/runtime/hashing/Crc32cJVM.kt

```diff
@@ -0,0 +1,664 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package aws.smithy.kotlin.runtime.hashing
+
+import java.util.zip.Checksum
+import kotlin.experimental.and
+import kotlin.experimental.xor
+
+public actual class Crc32c : Crc32cBase() {
+    private val md = CRC32CImpl()
+
+    override fun update(input: ByteArray, offset: Int, length: Int): Unit = md.update(input, offset, length)
+
+    override fun digest(): ByteArray {
+        val x = digestValue()
+        reset()
+        return byteArrayOf(
+            ((x shr 24) and 0xffu).toByte(),
```

Comment:
Why are we changing endianess here? I'm not sure we actually consume this anywhere but I'm curious where this is stemming from

Reply:
I initially copied this method from Crc32, which [says that the .digest() byte array should be in big-endian form](https://github.com/awslabs/smithy-kotlin/blob/main/runtime/hashing/common/src/aws/smithy/kotlin/runtime/hashing/Crc32.kt#L18)

big-endian means that the most significant byte should be placed in the array first, and the rest of the bytes should be placed in order of decreasing significance. we should use `shr` to accomplish that behavior.

___
example:

The CRC32C value of "foobar" is 224353407 as an unsigned integer, which is `0000 1101 0101 1111 0101 1100 0111 1111` in binary. With `0000 1101 0101 1111 0101 1100 0111 1111 shl 24 and 0xffu`, the first element in the byte array becomes 0, but it should be the top byte `0000 1101`, which is 13.

Reply:
Right I wasn't questioning the implementation, you are correct that your change results in big-endian. I was just questioning _why_ we were changing the endianess of the return value. Is there a spec somewhere dictating how the CRC value should be encoded or are we just arbitrarily making a decision here where either big/little endian would be fine and you were just updating to reflect the comment?

Reply:
I wasn't considering any spec when making the change, just updating to reflect the comment. Both little-endian or big-endian would work, but the original configuration with `shl` was neither because we were truncating the top 24 bits